spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarTemplate.java

```diff
@@ -42,13 +44,25 @@
 
 	private final PulsarProducerFactory<T> producerFactory;
 
+	private final List<ProducerInterceptor> interceptors;
+
 	/**
```

Comment:
> I'm not sure if the way I implemented it now is the 'Spring-way' because I was previously only using the framework, not developing it.

The one change I would make, is add the injection of these interceptors into the PulsarAutoConfiguration as such:
```java
@Bean
@ConditionalOnMissingBean(PulsarTemplate.class)
public PulsarTemplate<?> pulsarTemplate(PulsarProducerFactory<?> pulsarProducerFactory, ObjectProvider<ProducerInterceptor> interceptors) {
  return new PulsarTemplate<>(pulsarProducerFactory, interceptors.orderedStream().toList());
}
```
You will see this pattern prevalent in the Framework and especially in Boot when it comes to injecting one-or-more optional beans. The beauty of it is that it provides a nice stream API for lazy retrieving the beans. 
As you can see, the `interceptors.orderedStream().toList());` above will give an ordered list of any PI beans in the app context. 

Consider a user defining these interceptors beans:

```java
@Bean
@Order(200)
ProducerInterceptor interceptorFoo() {
  return null;
}

@Bean
@Order(100)
ProducerInterceptor interceptorBar() {
  return null;
}
```

then the list would contain `interceptorBar, interceptorFoo` as it respects the bean order.